lib/nexpose/vulnerability.rb

```diff
@@ -112,8 +112,10 @@ def add_bc_to_ssl_cipher_list(source)
     def cleanup_html(source)
       result = source.to_s
       result.gsub!(/<ContainerBlockElement>(.*?)<\/ContainerBlockElement>/m){|m| "#{ $1 }"}
-      result.gsub!(/<Paragraph preformat=\"true\">(\s*)<Paragraph preformat=\"true\">(.*?)<\/Paragraph>(\s*)<\/Paragraph>/mi){|m| "\nbc. #{ $2 }\n\n"}
-      result.gsub!(/<Paragraph preformat=\"true\">(.*?)<\/Paragraph>/mi){|m| "\nbc. #{ $1 }\n\n"}
+      result.gsub!(/<Paragraph preformat=\"true\">(\s*)<Paragraph preformat=\"true\">(.*?)<\/Paragraph>(\s*)<\/Paragraph>/i){"@#{$2}@"}
+      result.gsub!(/<Paragraph preformat=\"true\">(.*?)<\/Paragraph>/i){"@#{$1}@"}
+      result.gsub!(/<Paragraph preformat=\"true\">(\s*)<Paragraph preformat=\"true\">(.*?)<\/Paragraph>(\s*)<\/Paragraph>/mi){|m| "\nbc.. #{ $2 }\n\np. "}
+      result.gsub!(/<Paragraph preformat=\"true\">(.*?)<\/Paragraph>/mi){|m| "\nbc.. #{ $1 }\n\np. "}
```

Comment:
What about this:

```suggestion
      result.gsub!(/<Paragraph preformat=\"true\">(\s*)<Paragraph preformat=\"true\">(.*?)<\/Paragraph>(\s*)<\/Paragraph>/mi) do
        text = $2
        text[/\n/] ? "\nbc.. #{ text }\n\np. " : "@#{text}@"
      end
      result.gsub!(/<Paragraph preformat=\"true\">(.*?)<\/Paragraph>/mi) do
        text = $1
        text[/\n/] ? "\nbc.. #{ text }\n\np. " : "@#{text}@"
      end
```

Basically the idea is to reuse the same regex but check the text for newlines. If the text has newlines (` text[/\n/] `) replace with code blocks (`"\nbc.. #{ text }\n\np. " `) otherwise replace with inline (`"@#{text}@"`)